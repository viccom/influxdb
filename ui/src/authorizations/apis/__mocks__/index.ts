import {authorization} from './data'

// Types
import {Authorization} from 'src/api'

export const getAuthorizations = async (): Promise<Authorization[]> => {
  return Promise.resolve([authorization, {...authorization, id: '1'}])
}
