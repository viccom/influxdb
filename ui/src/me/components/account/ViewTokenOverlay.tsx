// Libraries
import React, {PureComponent} from 'react'

// Components
import {OverlayContainer, OverlayBody, OverlayHeading} from 'src/clockface'
import PermissionsWidget, {
  PermissionsWidgetMode,
  PermissionsWidgetSelection,
} from 'src/shared/components/permissionsWidget/PermissionsWidget'
import CopyText from 'src/shared/components/CopyText'

// Types
import {Authorization, Permission} from 'src/api'

// Actions
import {NotificationAction} from 'src/types'

const {Write, Read} = Permission.ActionEnum

interface Props {
  onNotify: NotificationAction
  auth: Authorization
  onDismissOverlay: () => void
}

const actions = [Read, Write]

export default class ViewTokenOverlay extends PureComponent<Props> {
  public render() {
    const {description, permissions} = this.props.auth
    const {onNotify} = this.props

    return (
      <OverlayContainer>
        <OverlayHeading title={description} onDismiss={this.handleDismiss} />
        <OverlayBody>
          <CopyText copyText={this.props.auth.token} notify={onNotify} />
          <PermissionsWidget
            mode={PermissionsWidgetMode.Read}
            heightPixels={500}
          >
            {permissions.map((p, i) => {
              return (
                <PermissionsWidget.Section
                  key={i}
                  id={this.id(p)}
                  title={this.title(p)}
                  mode={PermissionsWidgetMode.Read}
                >
                  {actions.map((a, i) => (
                    <PermissionsWidget.Item
                      key={i}
                      id={this.itemID(p, a)}
                      label={a}
                      selected={this.selected(p, a)}
                    />
                  ))}
                </PermissionsWidget.Section>
              )
            })}
          </PermissionsWidget>
        </OverlayBody>
      </OverlayContainer>
    )
  }

  private selected = (
    permission: Permission,
    action: Permission.ActionEnum
  ): PermissionsWidgetSelection => {
    if (permission.action === action) {
      return PermissionsWidgetSelection.Selected
    }

    return PermissionsWidgetSelection.Unselected
  }

  private itemID = (
    permission: Permission,
    action: Permission.ActionEnum
  ): string => {
    return `${permission.resource.type}-${action}-${permission.resource.id ||
      '*'}-${permission.resource.orgID || '*'}`
  }

  private id = (permission: Permission): string => {
    return permission.resource.type
  }

  private title = (permission: Permission): string => {
    const org = permission.resource.org || '*'
    const name = permission.resource.name || '*'
    const type = permission.resource.type

    if (permission.resource.name || permission.resource.org) {
      return `${org}:${type}:${name}`
    }

    return `${permission.resource.type}:*`
  }

  private handleDismiss = () => {
    this.props.onDismissOverlay()
  }
}
