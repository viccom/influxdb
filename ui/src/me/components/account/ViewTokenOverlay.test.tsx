// Libraries
import React from 'react'
import {mount} from 'enzyme'

// Components
import ViewTokenOverlay from 'src/me/components/account/ViewTokenOverlay'

// Fixtures
import {authorization as auth} from 'src/authorizations/apis/__mocks__/data'

const setup = (override?) => {
  const props = {
    auth,
    ...override,
  }

  const wrapper = mount(<ViewTokenOverlay {...props} />)

  return {wrapper}
}

describe('Account', () => {
  describe('rendering', () => {
    it('renders!', () => {
      const {wrapper} = setup()

      expect(wrapper.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
