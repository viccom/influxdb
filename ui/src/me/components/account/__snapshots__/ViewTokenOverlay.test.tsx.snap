// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account rendering renders! 1`] = `
<ViewTokenOverlay
  auth={
    Object {
      "description": "im a token",
      "id": "030444b11fb10000",
      "links": Object {
        "self": "/api/v2/authorizations/030444b11fb10000",
        "user": "/api/v2/users/030444b10a710000",
      },
      "orgID": "030444b10a713000",
      "permissions": Array [
        Object {
          "action": "write",
          "resource": Object {
            "type": "orgs",
          },
        },
        Object {
          "action": "write",
          "resource": Object {
            "type": "buckets",
          },
        },
      ],
      "status": "active",
      "token": "ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA==",
      "user": "watts",
      "userID": "030444b10a710000",
    }
  }
>
  <OverlayContainer
    maxWidth={800}
  >
    <div
      className="overlay--container"
      style={
        Object {
          "maxWidth": "800px",
        }
      }
    >
      <OverlayHeading
        onDismiss={[Function]}
        title="im a token"
      >
        <div
          className="overlay--heading"
        >
          <div
            className="overlay--title"
          >
            im a token
          </div>
          <button
            className="overlay--dismiss"
            onClick={[Function]}
          />
        </div>
      </OverlayHeading>
      <OverlayBody>
        <div
          className="overlay--body"
        >
          <CopyText
            copyText="ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA=="
          >
            <div
              className="script-snippet--container"
            >
              <FancyScrollbar
                autoHeight={true}
                autoHide={false}
                maxHeight={400}
                setScrollTop={[Function]}
                style={Object {}}
              >
                <Scrollbars
                  autoHeight={true}
                  autoHeightMax={400}
                  autoHeightMin={0}
                  autoHide={false}
                  autoHideDuration={250}
                  autoHideTimeout={1000}
                  className="fancy-scroll--container"
                  hideTracksWhenNotNeeded={false}
                  onScroll={[Function]}
                  renderThumbHorizontal={[Function]}
                  renderThumbVertical={[Function]}
                  renderTrackHorizontal={[Function]}
                  renderTrackVertical={[Function]}
                  renderView={[Function]}
                  style={Object {}}
                  tagName="div"
                  thumbMinSize={30}
                  universal={false}
                >
                  <div
                    className="fancy-scroll--container"
                    style={
                      Object {
                        "height": "auto",
                        "maxHeight": 400,
                        "minHeight": 0,
                        "overflow": "hidden",
                        "position": "relative",
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      className="fancy-scroll--view"
                      key="view"
                      style={
                        Object {
                          "WebkitOverflowScrolling": "touch",
                          "bottom": undefined,
                          "left": undefined,
                          "marginBottom": 0,
                          "marginRight": 0,
                          "maxHeight": 400,
                          "minHeight": 0,
                          "overflow": "scroll",
                          "position": "relative",
                          "right": undefined,
                          "top": undefined,
                        }
                      }
                    >
                      <p>
                        ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA==
                      </p>
                    </div>
                    <div
                      className="fancy-scroll--track-h"
                      key="trackHorizontal"
                      style={
                        Object {
                          "display": "none",
                          "height": 6,
                          "position": "absolute",
                        }
                      }
                    >
                      <div
                        className="fancy-scroll--thumb-h"
                        style={
                          Object {
                            "display": "block",
                            "height": "100%",
                            "position": "relative",
                          }
                        }
                      />
                    </div>
                    <div
                      className="fancy-scroll--track-v"
                      key="trackVertical"
                      style={
                        Object {
                          "display": "none",
                          "position": "absolute",
                          "width": 6,
                        }
                      }
                    >
                      <div
                        className="fancy-scroll--thumb-v"
                        style={
                          Object {
                            "display": "block",
                            "position": "relative",
                            "width": "100%",
                          }
                        }
                      />
                    </div>
                  </div>
                </Scrollbars>
              </FancyScrollbar>
              <CopyToClipboard
                onCopy={[Function]}
                text="ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA=="
              >
                <Button
                  active={false}
                  color="default"
                  customClass="copy-button"
                  onClick={[Function]}
                  shape="none"
                  size="sm"
                  status="default"
                  text="Copy"
                  titleText="copy to clipboard"
                  type="button"
                >
                  <button
                    className="button button-sm button-default copy-button"
                    disabled={false}
                    onClick={[Function]}
                    tabIndex={0}
                    title="copy to clipboard"
                    type="button"
                  >
                    Copy
                  </button>
                </Button>
              </CopyToClipboard>
            </div>
          </CopyText>
          <PermissionsWidget
            heightPixels={500}
            mode="read"
          >
            <div
              className="permissions-widget"
              style={
                Object {
                  "height": "500px",
                }
              }
            >
              <div
                className="permissions-widget--header"
              >
                Summary of access permissions
              </div>
              <div
                className="permissions-widget--body"
              >
                <FancyScrollbar
                  autoHeight={false}
                  autoHide={false}
                  maxHeight={null}
                  setScrollTop={[Function]}
                  style={Object {}}
                >
                  <Scrollbars
                    autoHeight={false}
                    autoHeightMax={null}
                    autoHeightMin={0}
                    autoHide={false}
                    autoHideDuration={250}
                    autoHideTimeout={1000}
                    className="fancy-scroll--container"
                    hideTracksWhenNotNeeded={false}
                    onScroll={[Function]}
                    renderThumbHorizontal={[Function]}
                    renderThumbVertical={[Function]}
                    renderTrackHorizontal={[Function]}
                    renderTrackVertical={[Function]}
                    renderView={[Function]}
                    style={Object {}}
                    tagName="div"
                    thumbMinSize={30}
                    universal={false}
                  >
                    <div
                      className="fancy-scroll--container"
                      style={
                        Object {
                          "height": "100%",
                          "overflow": "hidden",
                          "position": "relative",
                          "width": "100%",
                        }
                      }
                    >
                      <div
                        className="fancy-scroll--view"
                        key="view"
                        style={
                          Object {
                            "WebkitOverflowScrolling": "touch",
                            "bottom": 0,
                            "left": 0,
                            "marginBottom": 0,
                            "marginRight": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <PermissionsWidgetSection
                          id="orgs"
                          key=".$0"
                          mode="read"
                          title="orgs:*"
                        >
                          <section
                            className="permissions-widget--section"
                          >
                            <header
                              className="permissions-widget--section-heading"
                            >
                              <h3
                                className="permissions-widget--section-title"
                              >
                                orgs:*
                              </h3>
                            </header>
                            <ul
                              className="permissions-widget--section-list"
                            >
                              <PermissionsWidgetItem
                                id="orgs-read-*-*"
                                key=".$0"
                                label="read"
                                mode="read"
                                selected="unselected"
                              >
                                <li
                                  className="permissions-widget--item unselected"
                                  onClick={[Function]}
                                >
                                  <div
                                    className="permissions-widget--icon"
                                  >
                                    <span
                                      className="icon remove"
                                    />
                                  </div>
                                  <label
                                    className="permissions-widget--item-label"
                                  >
                                    read
                                  </label>
                                </li>
                              </PermissionsWidgetItem>
                              <PermissionsWidgetItem
                                id="orgs-write-*-*"
                                key=".$1"
                                label="write"
                                mode="read"
                                selected="selected"
                              >
                                <li
                                  className="permissions-widget--item selected"
                                  onClick={[Function]}
                                >
                                  <div
                                    className="permissions-widget--icon"
                                  >
                                    <span
                                      className="icon checkmark"
                                    />
                                  </div>
                                  <label
                                    className="permissions-widget--item-label"
                                  >
                                    write
                                  </label>
                                </li>
                              </PermissionsWidgetItem>
                            </ul>
                          </section>
                        </PermissionsWidgetSection>
                        <PermissionsWidgetSection
                          id="buckets"
                          key=".$1"
                          mode="read"
                          title="buckets:*"
                        >
                          <section
                            className="permissions-widget--section"
                          >
                            <header
                              className="permissions-widget--section-heading"
                            >
                              <h3
                                className="permissions-widget--section-title"
                              >
                                buckets:*
                              </h3>
                            </header>
                            <ul
                              className="permissions-widget--section-list"
                            >
                              <PermissionsWidgetItem
                                id="buckets-read-*-*"
                                key=".$0"
                                label="read"
                                mode="read"
                                selected="unselected"
                              >
                                <li
                                  className="permissions-widget--item unselected"
                                  onClick={[Function]}
                                >
                                  <div
                                    className="permissions-widget--icon"
                                  >
                                    <span
                                      className="icon remove"
                                    />
                                  </div>
                                  <label
                                    className="permissions-widget--item-label"
                                  >
                                    read
                                  </label>
                                </li>
                              </PermissionsWidgetItem>
                              <PermissionsWidgetItem
                                id="buckets-write-*-*"
                                key=".$1"
                                label="write"
                                mode="read"
                                selected="selected"
                              >
                                <li
                                  className="permissions-widget--item selected"
                                  onClick={[Function]}
                                >
                                  <div
                                    className="permissions-widget--icon"
                                  >
                                    <span
                                      className="icon checkmark"
                                    />
                                  </div>
                                  <label
                                    className="permissions-widget--item-label"
                                  >
                                    write
                                  </label>
                                </li>
                              </PermissionsWidgetItem>
                            </ul>
                          </section>
                        </PermissionsWidgetSection>
                      </div>
                      <div
                        className="fancy-scroll--track-h"
                        key="trackHorizontal"
                        style={
                          Object {
                            "display": "none",
                            "height": 6,
                            "position": "absolute",
                          }
                        }
                      >
                        <div
                          className="fancy-scroll--thumb-h"
                          style={
                            Object {
                              "display": "block",
                              "height": "100%",
                              "position": "relative",
                            }
                          }
                        />
                      </div>
                      <div
                        className="fancy-scroll--track-v"
                        key="trackVertical"
                        style={
                          Object {
                            "display": "none",
                            "position": "absolute",
                            "width": 6,
                          }
                        }
                      >
                        <div
                          className="fancy-scroll--thumb-v"
                          style={
                            Object {
                              "display": "block",
                              "position": "relative",
                              "width": "100%",
                            }
                          }
                        />
                      </div>
                    </div>
                  </Scrollbars>
                </FancyScrollbar>
              </div>
            </div>
          </PermissionsWidget>
        </div>
      </OverlayBody>
    </div>
  </OverlayContainer>
</ViewTokenOverlay>
`;
